### PR TITLE
Improvements to the Repository Settings

### PR DESCRIPTION
**Pull request by madoar**
This PR tries to improve some points from #667.
More detailed it adds the following features to the repositories list in the repositories settings:

- When viewing the repositories list in the repositories settings you can now see the priorities of the repositories, via a number in front of the repository
- The order of the repositories in the repository list is now changeable via drag and drop. The current behavior is, that directly after the order was changed the new repository order is saved to disk, maybe we want to change that and add a "Save"-button.

grafik

A still open problem is, that this PR breaks the text-color of the `ListCell`-content. I guess that this is because this PR changes the structure of the `ListCell`-content, which was previously simply a string. Now the text content is wrapped inside Labels, which are inside a GridPane, which is added to the ListCell as its graphic.

grafik

### CODE REVIEW

File: phoenicis-javafx/src/main/java/org/phoenicis/javafx/views/mainwindow/settings/DragableRepositoryListCell.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
+/**
+ *
```
Comment (qparis) [inline phoenicis-javafx/src/main/java/org/phoenicis/javafx/views/mainwindow/settings/DragableRepositoryListCell.java]: Please remove this

File: phoenicis-javafx/src/main/java/org/phoenicis/javafx/views/mainwindow/settings/DragableRepositoryListCell.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+/**
+ * @author marc
+ *
+ */
+public class DragableRepositoryListCell extends ListCell<String> implements ChangeListener<Number> {
+
+	/**
+	 * What should be done after this cell has been dragged
+	 */
+	private Runnable onDragDone;
```
Comment (qparis) [inline phoenicis-javafx/src/main/java/org/phoenicis/javafx/views/mainwindow/settings/DragableRepositoryListCell.java]: Can this be final

File: phoenicis-javafx/src/main/java/org/phoenicis/javafx/views/mainwindow/settings/DragableRepositoryListCell.java [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+import javafx.scene.layout.GridPane;
+
+/**
+ * @author marc
+ *
+ */
+public class DragableRepositoryListCell extends ListCell<String> implements ChangeListener<Number> {
+
+	/**
+	 * What should be done after this cell has been dragged
```
Comment (qparis) [inline phoenicis-javafx/src/main/java/org/phoenicis/javafx/views/mainwindow/settings/DragableRepositoryListCell.java]: This comment is not needed, however, the corresponding javadoc can be written on the constructor

**Comment by plata** [comment]
Why do we have numbers and an order? Isn't this duplicate? Also, I think that it is not clear what the meaning of the numbers is.
I would prefer to have just the order and add an explanatory text describing that you can change the priority by drag and drop.

**Comment by madoar** [comment]
The problem currently is, that the order of the List isn't visible to the user. What is the repository with the highest and the lowest priority?
To visualize this I've added a number that everyone can understand. I agree with you that we should add a short legend for this.

**Comment by madoar** [comment]
I've just fixed the css problem, and removed the center alignment of the priority.

About the priority information in general:
I'm not sure how to handle them in the final version. But I think we need to somehow show the user what priority which repository has. Therefore I would suggest merging this now and postponing the final decision how to do this to a later time.

**Comment by plata** [comment]
I would like to have the legend in this PR already. Should not be a big deal.

**Comment by madoar** [comment]
I was just too lazy to do that^^
grafik

grafik

**Comment by madoar** [comment]
Just as a note: I think somewhere between the file reading and the loading of the repositories the list gets reversed. We should check where that happens (and if it really happens...), because I don't think we should reverse the list, this only makes working with the repository list more complex.

**Pull request merged**